src/basic/terminal-util.c

```diff
@@ -857,6 +857,41 @@ unsigned lines(void) {
         return cached_lines;
 }
 
+int terminal_set_size_fd(int fd, const char *ident, unsigned rows, unsigned cols) {
+        struct winsize ws;
+
+        if (rows == UINT_MAX && cols == UINT_MAX)
+                return 0;
+
+        if (ioctl(fd, TIOCGWINSZ, &ws) < 0)
+                return log_debug_errno(errno,
+                                       "TIOCGWINSZ ioctl for getting %s size failed, not setting terminal size: %m",
+                                       ident ?: "TTY");
+
+        if (rows == UINT_MAX)
+                rows = ws.ws_row;
+        if (cols == UINT_MAX)
+                cols = ws.ws_col;
+
+        if (rows > USHRT_MAX)
+                rows = USHRT_MAX;
+        if (cols > USHRT_MAX)
+                cols = USHRT_MAX;
+
+        if (rows == ws.ws_row && cols == ws.ws_col)
+                return 0;
+
+        if (rows != ws.ws_row)
```

Comment:
you can drop this if check. after all if it the two values are equal then assigning one to the other is a NOP